jerry-core/api/jerry.c

```diff
@@ -3661,6 +3661,184 @@ jerry_foreach_object_property (const jerry_value_t obj_val, /**< object value */
+      uint32_t index = ecma_string_get_array_index (key_p);
+
+      /* Step 2. Filter by key type */
+      if (filter & (JERRY_PROPERTY_FILTER_EXLCUDE_STRINGS
+                    | JERRY_PROPERTY_FILTER_EXLCUDE_SYMBOLS
+                    | JERRY_PROPERTY_FILTER_EXLCUDE_INTEGER_INDICES))
+      {
+        if (ecma_is_value_symbol (key))
+        {
+          if (filter & JERRY_PROPERTY_FILTER_EXLCUDE_SYMBOLS)
+          {
+            continue;
+          }
+        }
+        else if (index != ECMA_STRING_NOT_ARRAY_INDEX)
+        {
+          if ((filter & JERRY_PROPERTY_FILTER_EXLCUDE_INTEGER_INDICES)
+              || ((filter & JERRY_PROPERTY_FILTER_EXLCUDE_STRINGS)
+                  && !(filter & JERRY_PROPERTY_FILTER_INTEGER_INDICES_AS_NUMBER)))
+          {
+            continue;
+          }
+        }
+        else if (filter & JERRY_PROPERTY_FILTER_EXLCUDE_STRINGS)
+        {
+          continue;
+        }
+      }
+
+      /* Step 3. Filter property attributes */
+      if (filter & (JERRY_PROPERTY_FILTER_EXLCUDE_NON_CONFIGURABLE
+                    | JERRY_PROPERTY_FILTER_EXLCUDE_NON_ENUMERABLE
+                    | JERRY_PROPERTY_FILTER_EXLCUDE_NON_WRITABLE))
+      {
+        ecma_property_descriptor_t prop_desc;
+        ecma_value_t status = ecma_op_object_get_own_property_descriptor (obj_iter_p, key_p, &prop_desc);
+
+#if ENABLED (JERRY_BUILTIN_PROXY)
+        if (ECMA_IS_VALUE_ERROR (status))
+        {
+          ecma_collection_free (prop_names_p);
+          ecma_collection_free (result_p);
+          return jerry_throw (ECMA_VALUE_ERROR);
+        }
+#endif /* ENABLED (JERRY_BUILTIN_PROXY) */
+
+        JERRY_ASSERT (ecma_is_value_true (status));
+        uint16_t flags = prop_desc.flags;
+        ecma_free_property_descriptor (&prop_desc);
+
+        if ((!(flags & ECMA_PROP_IS_CONFIGURABLE)
+             && (filter & JERRY_PROPERTY_FILTER_EXLCUDE_NON_CONFIGURABLE))
+            || (!(flags & ECMA_PROP_IS_ENUMERABLE)
+                && (filter & JERRY_PROPERTY_FILTER_EXLCUDE_NON_ENUMERABLE))
+            || (!(flags & ECMA_PROP_IS_WRITABLE)
+                && (filter & JERRY_PROPERTY_FILTER_EXLCUDE_NON_WRITABLE)))
+        {
+          continue;
+        }
+      }
+
+      if (index != ECMA_STRING_NOT_ARRAY_INDEX
+          && (filter & JERRY_PROPERTY_FILTER_INTEGER_INDICES_AS_NUMBER))
+      {
+        ecma_deref_ecma_string (key_p);
+        key = ecma_make_uint32_value (index);
+      }
+      else
+      {
+        ecma_ref_ecma_string (key_p);
+      }
+
+      if ((filter & JERRY_PROPERTY_FILTER_TRAVERSE_PROTOTYPE_CHAIN) && obj_iter_p != obj_p)
+      {
+        uint32_t duplicate_idx = 0;
+        for (; duplicate_idx < result_p->item_count; duplicate_idx++)
+        {
+          ecma_value_t value = result_p->buffer_p[duplicate_idx];
+          JERRY_ASSERT (ecma_is_value_prop_name (value) || ecma_is_value_number (value));
+          if (ecma_is_value_number (value))
```

Comment:
`JERRY_UNLIKELY ()` or `!ecma_is_value_number (value)` and reverse the conditions